lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -323,18 +323,19 @@ void LocalEnforcer::execute_actions(
 void LocalEnforcer::handle_activate_service_action(
     SessionMap& session_map, const std::unique_ptr<ServiceAction>& action_p,
     SessionUpdate& session_update) {
-  const std::string imsi                  = action_p->get_imsi(),
-                    msisdn                = action_p->get_msisdn(),
-                    ip_addr               = action_p->get_ip_addr(),
-                    ipv6_addr             = action_p->get_ipv6_addr();
-  const Teids teids                       = action_p->get_teids();
-  const auto ambr                         = action_p->get_ambr();
-  const std::vector<PolicyRule> rule_defs = action_p->get_rule_definitions();
+  const std::string imsi      = action_p->get_imsi(),
+                    msisdn    = action_p->get_msisdn(),
+                    ip_addr   = action_p->get_ip_addr(),
+                    ipv6_addr = action_p->get_ipv6_addr();
+  const Teids teids           = action_p->get_teids();
+  const auto ambr             = action_p->get_ambr();
+  RulesToProcess to_process;
+  to_process.rules = action_p->get_rule_definitions();
```

Comment:
will `ServiceAction` be updated to use RulesToProcess too?

Or in other words, who will set the versions of those rules? 
Is this versioning only needed to make sure sessiond and pipelined are in sync? Or is policyd also impacted by that versioning?
(sorry if you have already described this somewhere else )

Reply:
I think I can leave ServiceAction alone as processing that information is handled in `LocalEnforcer`. When it activates/deactivates rules in `SessionState`, we will fetch / populate the rule version.

The rule versioning will be owned by SessionD. So whenever it activates / deactivates a rule, SessionD will keep track of the updated version number. PipelineD will report the version attached to rule stats as well. 

For example if ruleX is installed with version 1 and then removed with version 2. If PipelineD keeps sending stats for ruleX at version 1, we know the deactivation didn't go through properly.
So this versioning exchange will help us keep track of the state syncs better. 

This is for `PipelineD` and `SessionD` only.

Reply:
Versioning make sense. 

We just have to make sure we don't fix pipelined issues at sessiond level

Reply:
Yeah totally agree.